apps/users/migrations/0004_alter_user_email_field.py

```diff
@@ -1,24 +1,40 @@
 # Generated by Django 5.1.7 on 2025-03-19 15:05
 
-from django.db import migrations, models
+from django.db import connection, migrations, models
 
 
 # Support for CIEmailField was be depreacated after Django 4.2 in favor
 # of vanilla EmailField. Below is how to migrate to the vanilla EmailField whilst
 # creating a non-deterministic collation. See Adam's post for more information
 # https://adamj.eu/tech/2023/02/23/migrate-django-postgresql-ci-fields-case-insensitive-collation/
+# further discussion on dropping index when altering a field here:
+# https://code.djangoproject.com/ticket/33901#comment:18
+def drop_index_if_exsists(apps, schema_editor):
+    """ Removes auto-generated index from 0002_case_insensitive_email migration """
+    with connection.cursor() as cursor:
+        cursor.execute("SELECT indexname FROM pg_indexes WHERE indexname LIKE 'users_user_email%like';")
+        result = cursor.fetchone()
+        if result:
+            cursor.execute(f"DROP INDEX IF EXISTS {result[0]};")
+
 class Migration(migrations.Migration):
 
     dependencies = [
         ("users", "0003_add_first_last_names"),
+        ("hip", "0025_case_insensitive_collation"),
     ]
 
     operations = [
+        migrations.RunPython(drop_index_if_exsists, migrations.RunPython.noop),
         migrations.AlterField(
             model_name="user",
             name="email",
             field=models.EmailField(
                 db_collation="case_insensitive", max_length=255, unique=True
             ),
         ),
+        migrations.RunSQL(
+            sql=f'CREATE INDEX "users_user_email_idx" ON "users_user" ("email")',
+            reverse_sql='DROP INDEX IF EXISTS "users_user_email_idx"',
+        ),
```

Comment:
It seems that the database is unhappy when you use `AlterField` when changing from CIEmail to Vanilla even with the collation.  And that this error was thrown `django.db.utils.NotSupportedError: nondeterministic collations are not supported for operator class "varchar_pattern_ops"` ...Reading a bit about this in this thread here:

https://code.djangoproject.com/ticket/33901#comment:18

An alternative solution would have been to drop the email field, do a data migration and then add a new field, but that seemed more dangerous.